plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+RSpec.describe "Quoting chat message transcripts", type: :system, js: true do
+  fab!(:current_user) { Fabricate(:user) }
+  fab!(:chat_channel_1) { Fabricate(:chat_channel) }
+  fab!(:chat_message_1) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_2) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_3) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_4) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_5) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:topic) { Fabricate(:with_posts_topic) }
+  let(:last_topic_post) { topic.posts.last }
+
+  let(:chat_page) { PageObjects::Pages::Chat.new }
+  let(:chat_channel_page) { PageObjects::Pages::ChatChannel.new }
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+
+  before do
+    chat_system_bootstrap(current_user, [chat_channel_1])
+    sign_in(current_user)
+  end
+
+  def select_message(message, mobile: false)
+    if page.has_css?(".chat-message-container.selecting-messages")
+      chat_channel_page.message_by_id(message.id).find(".chat-message-selector").click
+    else
+      # we long press instead of hover on mobile
+      if mobile
+        chat_channel_page.message_by_id(message.id).click(delay: 0.5)
+      else
+        chat_channel_page.message_by_id(message.id).hover
+      end
+
+      # we also have a different actions menu on mobile
+      if mobile
+        find(".chat-message-action-item[data-id=\"selectMessage\"]").click
+      else
+        expect(page).to have_css(".chat-message-actions .more-buttons")
+        find(".chat-message-actions .more-buttons").click
+        find(".select-kit-row[data-value=\"selectMessage\"]").click
+      end
+    end
+  end
+
+  def cdp_allow_clipboard_access!
```

Comment:
This looks a little crazy, so an explanation of what I have found here. I found mention of this first on these pages:

- [Clipboard reading in Capybara](https://gist.github.com/kaisumi/933c5dd1b4486628d24109020cd808ca)
- [Execute CDP from a Capybara test - Today I Learned](https://til.hashrocket.com/posts/wuz98lrv5n-execute-cdp-from-a-capybara-test-)
- https://www.selenium.dev/documentation/webdriver/bidirectional/chrome_devtools/

CDP is the Chrome DevTools protocol https://chromedevtools.github.io/devtools-protocol/ which allows RPCs to control Chrome. There are tons of different things you can do with it https://chromedevtools.github.io/devtools-protocol/tot/Browser/ , for example setting permissions which is needed for clipboard access. The permissions that can be set are at https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/permissions/permission_descriptor.idl  . There is even crazy stuff in there like forcing a crash of the browser GPU process https://chromedevtools.github.io/devtools-protocol/tot/Browser/#method-crashGpuProcess .

The main downside is that if we ever want to add Firefox system tests this is not there, though Mozilla are gradually adding parity https://firefox-source-docs.mozilla.org/remote/cdp/ . 

What do you think of this? I like it because the clipboard-related stuff finally works, though I could see if we didn't want this added complexity in system tests.

Reply:
Im totally fine with it. Worst case if we ever want to support firefox and can't find a sane way to do this specific test with it, we can have example level metadata "chrome_only: true" to avoid the issue.

Reply:
Another note is that I value very much having very strong tests for things like notifications otherwise it's a pain to test and improve, so even if it would be running only on chrome that would still have tremendous value.

Reply:
Okay cool good to hear :+1: